src/tests/Microsoft.PowerFx.Core.Tests.Shared/ExpressionTestCases/JSON_OnlyTopLevel.txt

```diff
@@ -0,0 +1,142 @@
+#SETUP: EnableJsonFunctions
+
+>> JSON(Blank(), JSONFormat.OnlyOneLevel)
+"null"
+
+>> JSON(If(1<0,"true",{a:1}), JSONFormat.OnlyOneLevel)
+Errors: Error 5-25: The JSON function cannot serialize objects of type 'Void'.
+
+>> JSON("hello", JSONFormat.OnlyOneLevel)
+"""hello"""
+
+>> JSON(1.23, JSONFormat.OnlyOneLevel)
+"1.23"
+
+>> JSON(0, JSONFormat.OnlyOneLevel)
+"0"
+
+>> JSON(1, JSONFormat.OnlyOneLevel)
+"1"
+
+>> JSON(-1, JSONFormat.OnlyOneLevel)
+"-1"
+
+>> JSON(true, JSONFormat.OnlyOneLevel)
+"true"
+
+>> JSON(false, JSONFormat.OnlyOneLevel)
+"false"
+
+>> JSON("Back\slash", JSONFormat.OnlyOneLevel)
+"""Back\\slash"""
+
+>> JSON("Quotes "" and ' may be escaped", JSONFormat.OnlyOneLevel)
+"""Quotes \"" and ' may be escaped"""
+
+>> JSON($"Other {Char(13)} escaped {Char(10)} chars {Char(4)} should {Char(9)} also {Char(8)} be {Char(1)} escaped {Char(20)} properly", JSONFormat.OnlyOneLevel)
+"""Other \r escaped \n chars \u0004 should \t also \b be \u0001 escaped \u0014 properly"""
+
+// In \uXXXX escapings, letters of hex characters are upper case while they are lower case in Power Apps.
+// Technically, this is the exactly the same character.
+>> JSON(Concat(Sequence(128), Char(Value)), JSONFormat.OnlyOneLevel)
+"""\u0001\u0002\u0003\u0004\u0005\u0006\u0007\b\t\n\u000B\f\r\u000E\u000F\u0010\u0011\u0012\u0013\u0014\u0015\u0016\u0017\u0018\u0019\u001A\u001B\u001C\u001D\u001E\u001F !\""#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~\u007F\u0080"""
+
+>> JSON(RGBA(10, 30, 50, 0.75), JSONFormat.OnlyOneLevel)
+"""#0a1e32bf"""
+
+>> JSON(Color.Turquoise, JSONFormat.OnlyOneLevel)
+"""#40e0d0ff"""
+
+>> JSON(TimeUnit.Hours, JSONFormat.OnlyOneLevel)
+"""hours"""
+
+>> JSON(Date(2022,8,7), JSONFormat.OnlyOneLevel)
+"""2022-08-07"""
+
+>> JSON(DateTimeValue("1970-01-01T00:00:00Z"), JSONFormat.OnlyOneLevel)
+"""1970-01-01T00:00:00.000Z"""
+
+// Independent from local timezone
+>> With({dt: DateTime(1987,6,5,4,30,0)}, JSON(DateAdd(dt,-TimeZoneOffset(dt),TimeUnit.Minutes), JSONFormat.IndentFour & JSONFormat.OnlyOneLevel))
+"""1987-06-05T04:30:00.000Z"""
+
+>> JSON({a:1,b:"hello",c:true}, JSONFormat.OnlyOneLevel)
+"{""a"":1,""b"":""hello"",""c"":true}"
+
+>> JSON({a:1,b:"hello",c:true}, JSONFormat.Compact & JSONFormat.OnlyOneLevel)
+"{""a"":1,""b"":""hello"",""c"":true}"
+
+>> Substitute(Substitute(Substitute(JSON({a:1,b:"hello",c:true}, JSONFormat.IndentFour & JSONFormat.OnlyOneLevel), Char(10), "*"), Char(13), ""), "    ", "<t>")
+"{*<t>""a"": 1,*<t>""b"": ""hello"",*<t>""c"": true*}"
+
+>> JSON(Table({a:1,b:"hello",c:true,e:Blank()},{a:-3,b:"world",c:false,d:GUID("01234567-89AB-CDEF-0123-456789ABCDEF")}), JSONFormat.OnlyOneLevel)
+"[{},{}]"
```

Comment:
>[{},{}] [](http://example.com/codeflow?start=1&length=7)

Going into a table also counts as one level? That doesn't sound very useful... That means that we would only be able to serialize unwrapped arrays of primitives - which we don't support in Power Fx

Reply:
Agreed, tables shouldn't count as one level
This will allow us to be consistent between
JSON(First(table))
JSON(FirstN(table, n))